apps/studio/src/features/mail/service.ts

```diff
@@ -0,0 +1,40 @@
+import type { InvitationEmailTemplateData } from "./templates"
+import { createBaseLogger } from "~/lib/logger"
+import { isValidEmail } from "~/utils/email"
+import { sendMail } from "../../lib/mail"
+import { templates } from "./templates/templates"
+
+const logger = createBaseLogger({ path: "lib/mail/service" })
+
+export async function sendInvitation(
+  email: string,
+  data: InvitationEmailTemplateData,
+): Promise<void> {
+  if (!isValidEmail(email)) {
```

Comment:
[FYI] no action required on your end, but a short coming of the `isValidEmail` function is that the email is still treated as just another string. we could make use of a `Tagged` type from `type-fest` to say then `email is Tagged<string, "email">` so we get type safety and don't have to repeat this check again, but not very important